### PR TITLE
Rspec integration tests for secret store

### PR DESCRIPTION
**Pull request by jakelandis**
The unit tests (already committed) are quite comprehensive and cover all of the known edge cases.  The tests here are closer to regression tests that ensures :

* configuration substitution works in a running instance
* settings substitution works in a running instance
* bad password and missing values will prevent startup

### CODE REVIEW

File: qa/integration/README.md [hunk 1]
```diff
@@ -78,5 +78,16 @@ The integration tests need to be run from MacOS or Linux.  However, the tests ma
 3. Create a corresponding `test_file_input_spec.rb` in `specs` folder and use the `fixtures` object to get all services, config etc. The `.yml` and rspec file has to be the same name for the settings to be picked up. You can start LS inside the tests and assume all external services have already been started.
 4. Write rspec code to validate.
 
+### Logstash Keystore
```
Comment (original-brownbear) [inline qa/integration/README.md]: @jakelandis maybe better put this explanation into `secret_store_spec.rb`? Not many people will ever think about going to the Readme for documentation on a specific test imo => much more helpful there right?

File: qa/integration/specs/secret_store_spec.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  it "expands secret store variables from config" do
+    test_env["TEST_ENV_PATH"] = test_path
+    test_env["LOGSTASH_KEYSTORE_PASS"] = "keystore_pa9454w3rd"
+    @logstash.env_variables = test_env
+    @logstash.start_background_with_config_settings(config_to_temp_file(@fixture.config), settings_dir)
+    @logstash.wait_for_logstash
+    Stud.try(num_retries.times, [StandardError, RSpec::Expectations::ExpectationNotMetError]) do
+      # 10 outputs in the correct path. 10 generated outputs, mypath, and the tags all come from the secret store
+      expect(IO.read(File.join(File.join(test_path, "mypath"), "logstash_secretstore_test.output")).gsub("\n", "")).to eq("Hello world! mytag1,mytag2.mytag3Hello world! mytag1,mytag2.mytag3Hello world! mytag1,mytag2.mytag3Hello world! mytag1,mytag2.mytag3Hello world! mytag1,mytag2.mytag3Hello world! mytag1,mytag2.mytag3Hello world! mytag1,mytag2.mytag3Hello world! mytag1,mytag2.mytag3Hello world! mytag1,mytag2.mytag3Hello world! mytag1,mytag2.mytag3")
```
Comment (original-brownbear) [inline qa/integration/specs/secret_store_spec.rb]: @jakelandis how about 
```rb
"Hello world! mytag1,mytag2.mytag3" * 10
```

instead? :D

**Comment by original-brownbear** [comment]
@jakelandis just two random points, looks good in general :)

File: qa/integration/specs/secret_store_spec.rb (hunk 2 above)
Comment (jakelandis) [reply qa/integration/specs/secret_store_spec.rb]: Neat trick!  

done.

File: qa/integration/README.md (hunk 1 above)
Comment (jakelandis) [reply qa/integration/README.md]: moved comments to spec.

**Comment by jakelandis** [comment]
@original-brownbear - mind taking another look ?

**Comment by original-brownbear** [comment]
@jakelandis LGTM :)
